py/emitinlinethumb.c

```diff
@@ -35,6 +35,12 @@
 
 #if MICROPY_EMIT_INLINE_THUMB
 
+#if !defined(__clang__) && defined(__GNUC__) && __GNUC__ > 10
```

Comment:
I don't actually know when this attribute was supported in gcc, I just picked 10 as a guess that should work fine.

Reply:
how about `#if __has_attribute(__nonstring__)`?

Reply:
Ah, I didn't know about that, thanks for the hint!

Now changed.